src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -42,7 +43,6 @@ public async Task Invoke(HttpContext context)
             //    // downstream middleware and filters rely on this
             //    context.Items[ScriptConstants.AzureFunctionsHostKey] = scriptHost;
             //}
-            SetRequestId(context.Request);
```

Comment:
Moving this up to the new middleware, where we had it in v1 in the pipeline. That ensures that it runs even if we're not invoking a function (i.e. for admin APIs)